ide/xml/src/org/netbeans/modules/xml/sync/DataObjectSyncSupport.java

```diff
@@ -115,10 +115,9 @@ public void addRepresentation(Representation rep) {
 
         if (rep.represents(Document.class)) {
             synchronized (reps) {
-                for (Iterator it = reps.iterator(); it.hasNext();) {
-                    Representation next = (Representation) it.next();
+                for (Representation next : reps) {
                     if (next.represents(FileObject.class)) {
-                        it.remove();
+                        reps.remove(next);
```

Comment:
This is dangerous and will most probably blow at runtime. You are modifying the collection you are iterating, which will result in a `ConcurrentModificationException`.

Reply:
I wasn't sure if that was correct or not.. It's in a synchronized code block, though I think the problem you described still exists because the iterator is unspecified if the underlying collection is modified while the iteration is in progress..

Anyways, I think I properly reverted it..